hooks/templatedatavalidator.js

```diff
@@ -0,0 +1,122 @@
+var fs = require('fs');
+var path = require('path');
+const { parseJamboConfig } = require('../commands/helpers/utils/jamboconfigutils');
+const { error } = require('../commands/helpers/utils/logger');
+/**
+ * Validates page template's data configuration
+ * 
+ * @param {Object} pageData data that gets passed into a page template
+ * @returns {boolean} false if validator should throw an error
+ */
+module.exports = function (pageData) {
+  const jamboConfig = parseJamboConfig();
+  const validatorResults = [
+    isGlobalConfigValid(pageData.global_config), 
+    isPageVerticalConfigValid(pageData, jamboConfig)
+  ];
+  const isValid = !validatorResults.some((result) => !result);
+  return isValid;
+}
+
+/**
+ * Validates global config for the page template
+ * 
+ * @param {Object} globalConfig 
+ * @returns {boolean}
+ */
+function isGlobalConfigValid(globalConfig) {
+  if (!globalConfig.experienceKey) {
+    error('Missing Info: no experienceKey found.');
+    return false;
+  }
+  return true;
+}
+
+/**
+ * Valivates vertical config for current page template
+ * 
+ * @param {Object} pageData 
+ * @param {Object} jamboConfig 
+ * @returns {boolean}
+ */
+function isPageVerticalConfigValid(pageData, jamboConfig) {
+  let isValid = true;
+  const themeDirectory = path.resolve(jamboConfig.dirs.themes, jamboConfig.defaultTheme);
+  Object.keys(pageData.verticalsToConfig).forEach(key => {
```

Comment:
sorry i didn't notice this earlier, but I think it would be a little simpler if we used Array.prototype.every instead of an isValid param and forEach loop (also in isAllVerticalConfigsValid as well)

Reply:
I think Array.prototype.every stops evaluating the array (short circuits) the first time it finds an element that does not satisfy the given requirement. I think we should keep the forEach loop. I should make the changes for where I use Array.prototype.some as well since it also short circuits.

Reply:
Updated to use Reduce() instead, I think it's a bit cleaner now. What do you think?

Reply:
ahh good catch!